### PR TITLE
Vhc 279 using GitHub graphql

### PR DESCRIPTION
**Pull request by vhcsilva**

### CODE REVIEW

File: interfaces/authentication.ts [hunk 1]
```diff
@@ -8,9 +8,16 @@ export interface IUser {
   accessToken?: string;
 }
 
+export interface IRepository {
```
Comment (moshmage) [inline interfaces/authentication.ts:11]: use "RepositoryInfo" avoid `I` for interfaces

File: interfaces/authentication.ts [hunk 2]
```diff
@@ -8,9 +8,16 @@ export interface IUser {
...
+  name: string;
+  nameWithOwner: string;
+  isFork: boolean;
+}
+
 export interface IWallet {
```
Comment (moshmage) [inline interfaces/authentication.ts:17]: Same here, avoid `I` for interfaces

File: package.json [hunk 3]
```diff
@@ -28,6 +28,7 @@
     "ganache:deploy": "node ganache/deploy.js"
   },
   "dependencies": {
+    "@octokit/graphql": "^4.8.0",
```
Comment (moshmage) [inline package.json:31]: was the other ockokit we were using removed?

File: x-hooks/use-octokit-graph.tsx [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+    const { additions, deletions } = response["repository"]["pullRequest"];
+
+    return additions + deletions;
+  }
+
+  async function getIssueOrPullRequestComments(repositoryPath:  string, id: number) {
+    const { owner, repo } = getOwnerRepoFrom(repositoryPath);
+
+    const response = await getAllPages(`
+        query Comments($repo: String!, $owner: String!, $id: Int!, $cursor: String) {
```
Comment (moshmage) [inline x-hooks/use-octokit-graph.tsx:123]: put these queries into a `graph-ql-queries` folder, and then import them and use them

**Review by moshmage — CHANGES_REQUESTED** [review]

File: package.json (hunk 3 above)
Comment (vhcsilva) [reply package.json:31]: @moshmage no, we are using it on BE

**Review by moshmage — APPROVED** [review]

File: helpers/object.ts [hunk 5]
```diff
@@ -0,0 +1,9 @@
+export const getPropertyRecursively = (property, data) => {
```
Comment (moshmage) [inline helpers/object.ts:1]: `getPropertyRecursively` has to explicitly take in a type that and return that same type,

`export function getProp<T = any>(prop: string, data = {}): T[keyof T]`

If this isn't possible, then _all variables that use this function must explicitly say what their type is_

File: pages/api/past-events/move-to-open/index.ts [hunk 6]
```diff
@@ -53,19 +56,41 @@ async function post(req: NextApiRequest, res: NextApiResponse) {
...
+        if (currentRepo !== `${owner}/${repo}`) {
+          currentRepo = `${owner}/${repo}`;
+
+          const repositoryDetails = await githubAPI(RepositoryQueries.Details, {
+            repo,
+            owner
+          });
+
+          labelId = 
+            repositoryDetails["repository"]["labels"]["nodes"].find(label => label.name.toLowerCase() === "draft")?.id;
```
Comment (moshmage) [inline pages/api/past-events/move-to-open/index.ts:82]: why are we using `[]` notation? can't we simply use `.notation` (dot notation?)

File: pages/api/past-events/redeem-issue/index.ts [hunk 7]
```diff
@@ -58,13 +61,21 @@ async function post(req: NextApiRequest, res: NextApiResponse) {
...
+
+        const issueDetails = await githubAPI(IssueQueries.Details, {
           repo,
-          issue_number: +issue.githubId,
-          state: "closed"
+          owner,
+          issueId: +issue.githubId
         });
+      
+        const issueGithubId = issueDetails["repository"]["issue"]["id"];
```
Comment (moshmage) [inline pages/api/past-events/redeem-issue/index.ts:73]: Same here, use dot notation

File: types/octokit.d.ts [hunk 8]
```diff
@@ -0,0 +1,5 @@
+export type GraphQlQueryResponseData = {
```
Comment (moshmage) [inline types/octokit.d.ts:1]: I'm betting that GraphQL provides their own types, look into it

File: x-hooks/use-merge-data.ts [hunk 9]
```diff
@@ -17,7 +15,7 @@ const OctoData = {};
 export default function useMergeData() {
   const { loadRepos, repoList } = useRepos();
   const db = useApi();
-  const octokit = useOctokit();
+  const octokit = null;
```
Comment (moshmage) [inline x-hooks/use-merge-data.ts:18]: ?

**Review by moshmage — CHANGES_REQUESTED** [review]

File: helpers/object.ts [hunk 10]
```diff
@@ -0,0 +1,9 @@
+export function getPropertyRecursively<T> (property: string, data = {}): T[keyof T] {
+  if (data[property]) return data[property];
+
+  let found = {};
+
+  for(const key in data) found = getPropertyRecursively(property, data[key]);
```
Comment (moshmage) [inline helpers/object.ts:6]: We need to make this for short-circuit itself, or we are going to load forever,

```javascript
if (data[property]) return data[property];
let found = null;
const keys = Object.keys(data);
for (let x = 0; x <= keys.length - 1 && !found; x++) {
  found = getPropertyRecursively(property, keys[x]);
}
return found;
```

File: pages/api/past-events/move-to-open/index.ts [hunk 11]
```diff
@@ -94,8 +121,8 @@ async function post(req: NextApiRequest, res: NextApiResponse) {
 export default async function MoveToOpen(req: NextApiRequest,
                                          res: NextApiResponse) {
   switch (req.method.toLowerCase()) {
-  case "post":
-    await post(req, res);
+  case "get":
```
Comment (moshmage) [inline pages/api/past-events/move-to-open/index.ts:124]: doesn't this endpoint change the state? why did we change it to a GET ?

**Review by moshmage — APPROVED** [review]

**Pull request merged**